### PR TITLE
Add a slide on generic lambdas, including explicit template parameters

### PR DESCRIPTION
**Pull request by bernhardmgruber**
Fixes: #180, #181

- [x] Merge #188 first.

### CODE REVIEW

File: talk/morelanguage/lambda.tex [hunk 1]
```diff
@@ -192,6 +192,61 @@
...
+      [=, *this](...) { use(*this); };
+    \end{cppcode*}
+    Allowed since \cpp17: the object at \mintinline{cpp}{*this} is captured by value.
+  \end{block}
+  Details in  \href{https://www.nextptr.com/tutorial/ta1430524603/capture-this-in-lambda-expression-timeline-of-change}{this blog post}.
+\end{frame}
+
+\begin{frame}[fragile]
+  \frametitlecpp[14]{Generic lambdas}
+  \begin{block}{Generic lambdas (aka. polymorphic lambdas)}
```
Comment (hageboeck) [inline talk/morelanguage/lambda.tex:224]:
```suggestion
  \begin{block}{Generic lambdas (aka.\ polymorphic lambdas)}
```
Just a nitpick. Latex needs to know that this doesn't end a sentence for best spacing.

File: talk/morelanguage/lambda.tex [hunk 2]
```diff
@@ -192,6 +192,61 @@
...
+    \begin{itemize}
+      \item The type of a lambda parameter may be \mintinline{cpp}{auto}.
```
Comment (hageboeck) [inline talk/morelanguage/lambda.tex:226]:
```suggestion
      \item The type of lambda parameters may be \mintinline{cpp}{auto}.
```

File: talk/morelanguage/lambda.tex [hunk 3]
```diff
@@ -192,6 +192,61 @@
...
+      \end{cppcode*}
+      \item The generated \mintinline{cpp}{operator()} becomes a template function:
+      \begin{cppcode*}{linenos=false,gobble=4}
+        template <typename T, typename U>
+        auto operator()(T a, U b) const { return a + b; }
+      \end{cppcode*}
+      \item The types of \mintinline{cpp}{a} and \mintinline{cpp}{b} may be different.
+    \end{itemize}
+  \end{block}
+  \begin{block}{Explicit template parameters}
```
Comment (hageboeck) [inline talk/morelanguage/lambda.tex:238]:
```suggestion
  \begin{block}{Explicit template parameters (\cpp20)}
```
I'm not sure this is legal in our latex template, but what I mean is to put the restriction in the box title.

**Review by hageboeck — APPROVED** [review]
All good. Just 3 small suggestions.

File: talk/morelanguage/lambda.tex (hunk 3 above)
Comment (bernhardmgruber) [reply talk/morelanguage/lambda.tex:238]: Yes, we can do that!

**Pull request merged**